google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -157,21 +186,36 @@ def _get_ephemeral(service, project, instance, pub_key):
         """
 
         if (
-            not isinstance(service, googleapiclient.discovery.Resource)
+            not isinstance(credentials, Credentials)
             or not isinstance(project, str)
             or not isinstance(instance, str)
             or not isinstance(pub_key, str)
         ):
             raise TypeError("Cannot take None as an argument.")
 
-        # TODO(ryachen@) Add checks to ensure service object is valid.
+        if not credentials.valid:
+            request = google.auth.transport.requests.Request()
+            credentials.refresh(request)
 
-        request = service.sslCerts().createEphemeral(
-            project=project, instance=instance, body={"public_key": pub_key}
+        bearer_token = credentials.token
+
+        headers = {
+            "Authorization": "Bearer %s" % bearer_token,
```

Comment:
I think `format()` is preferred over `%`.

I think bearer_token can be moved inline.